sgt-back-end/server.ts

```diff
@@ -0,0 +1,191 @@
+import pg from 'pg';
+import express from 'express';
+import { ClientError } from './client-error.js';
+import { errorMiddleware } from './error-middleware.js';
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    if (!Number.isInteger(gradeId) || gradeId <= 0) {
+      throw new ClientError(400, '"gradeId" must be a positive integer');
+    }
+
+    const sql = `
+      select *
+        from "grades"
+        where "gradeId" = $1
+    `;
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (!grade) {
+      throw new ClientError(404, `Cannot find grade with "gradeId" ${gradeId}`);
+    }
+    res.json(grade);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "grades"
+    `;
+    const result = await db.query(sql);
+    const grades = Object.values(result.rows);
+    if (!grades) {
+      throw new ClientError(
+        500,
+        `An unexpected error has occured with our database.`
+      );
+    }
+    res.json(grades);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+
+    if (name === undefined) {
+      throw new ClientError(400, '"name" is a required field.');
+    }
+
+    if (course === undefined) {
+      throw new ClientError(400, '"course" is a required field.');
+    }
+
+    if (!Number.isInteger(score) || score < 0 || score > 100) {
+      throw new ClientError(
+        400,
+        '"score" is a required field and must be between 0 and 100.'
+      );
+    }
+
+    const sql = `
+      INSERT INTO "grades" ("name", "course", "score")
+           VALUES ($1, $2, $3)
+           RETURNING *
+    `;
+    const params = [name, course, score];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (!grade) {
```

Comment:
This isn't actually necessary since `db.query` will throw if it is unable to return a result.

Also, a 500 error isn't a Client Error, so if you wanted to throw such an error you would just `throw new Error()`.

Fix this in several places.